### DISCUSSION TITLE
Dúvida sobre dependência externa no projeto Core da aplicação

### DISCUSSION THREAD

**Discussion by alxc89** [balta-io/forum#398] [Tira Dúvidas] [answered]
No curso Segurança em APIs ASP.NET com JWT e Bearer Authentication na aula Adicionando suporte ao Mediator, você menciona adicionar o Mediator como dependência no projeto Core, que é onde fica o Domínio da aplicação. Isso não seria um problema, tendo em vista arquitetura limpa onde diz que o domínio não deve ter dependências externas?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @alxc89, não vejo como um "problema", já que temos a dependência do Mediatr... é a mesma relação com os DataAnnotations por exemplo.

Daria para migrar os Handlers para camada de Infra e deixar apenas as interfaces no domínio, sem problemas, só que neste cenário, é uma "limpeza" desnecessária ao meu ver.

**Reply by alxc89** [reply]
Entendi, obrigado pelo retorno!
